### PR TITLE
Only install dependencies once for examples

### LINKED ISSUE #118: make lint test does glide install twice
`make lint test` calls into the makefile in examples/keyvalue, which does it's own dependency management. It would be nice if this was somehow merged, so that glide isn't called twice. I'd honestly just put the example makefile goals into the main makefile, including any extra dependencies. This makes `make lint test` much slower.

### PR DESCRIPTION
**Pull request by sectioneight**
Fixes #118

### CODE REVIEW

**Comment by sectioneight** [comment]
@uber-go/service-framework

File: examples/keyvalue/Makefile [hunk 1]
```diff
@@ -1,13 +1,4 @@
-.PHONY: deps
-deps:
-	@echo "Installing UberFx dependencies..."
-	$(ECHO_V)$(MAKE) -C ../.. libdeps
-	@echo "Installing thriftrw..."
-	$(ECHO_V)go install ../../vendor/go.uber.org/thriftrw
```
Comment (sectioneight) [inline examples/keyvalue/Makefile]: Yea, this was a mistake. Fixing now.

File: Makefile [hunk 2]
```diff
@@ -121,8 +117,18 @@ gendoc:
 		xargs -I% md-to-godoc -input=%
 
 .PHONY: clean
-clean::
+clean:
```
Comment (madhuravi) [inline Makefile]: Why change this to single colon?

Comment (sectioneight) [inline Makefile]: All the other targets are single. Double is useful if it's going to be extended by some other file, but these make rules are all self-contained, so semantically single-colon is correct. I think this was added by accident (or copy-pasted from some internal makefiles, which may have had some extensibility concerns we don't have here).

Comment (madhuravi) [inline Makefile]: Have we thought about setting up convenient wrappers for makefile targets? We have an internal repo that teams use today and include in their Makefiles.

**Comment by sectioneight** [comment]
I think people will continue to use the existing build infrastructure. Are
there pieces here you'd like to see extracted internally?

**Pull request merged**